blobs.go

```diff
@@ -75,6 +75,17 @@ type BlobDeleter interface {
 	Delete(ctx context.Context, dgst digest.Digest) error
 }
 
+// BlobEnumerator allows to list blobs in storage.
+type BlobEnumerator interface {
+	// Enumerate fills 'digests' with a lexicographically sorted list of
+	// digests up to the size of 'digests' and returns the value 'n' for the
+	// number of entries which were filled. 'last' contains an offset in the
+	// list. It's a string representation of digest. Empty string means 0
+	// offset. 'err' will be set to io.OEF if there are no more entries to
+	// obtain.
+	Enumerate(ctx context.Context, digests []digest.Digest, last string) (n int, err error)
```

Comment:
After some discussion, we decided that a lexicographical ordering makes no sense for blobs. Would it be a huge change to make this work with a callback?

I'm sorry to flip flop on this one. I think I had suggested that this follow the semantics of tags and repository enumeration. In those cases, lexicographic paginations has a lot of benefits, semantically and storage-wise.